### PR TITLE
Ds 3202

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

File: R/ParseUserEnteredTable.R [hunk 1]
```diff
@@ -168,11 +168,23 @@ parseAsVectorOrMatrix <- function(m, warn = FALSE)
...
         row.names.given <- isQStatistic(m[1,1])
 
+    # If input data are percentages then we can do more tests
+    .is_pct <- function(x) { tmp <- asNumericVector(x); return(!is.null(attr(tmp, "statistic"))) }
+
     # Handling ambiguous row/column labels
     if (is.na(row.names.given) || is.na(col.names.given))
     {
+        # If input data are percentages then we can do more tests
+        .is_pct <- function(x) { tmp <- asNumericVector(x); return(!is.null(attr(tmp, "statistic"))) }
```
Comment (JustinCCYap) [inline R/ParseUserEnteredTable.R]: Does this need to be defined again?

File: R/ParseUserEnteredTable.R [hunk 2]
```diff
@@ -168,11 +168,23 @@ parseAsVectorOrMatrix <- function(m, warn = FALSE)
     if (any(!isTextNumeric(m[-1,-1], allow.missing = TRUE)))
         row.names.given <- isQStatistic(m[1,1])
 
+    # If input data are percentages then we can do more tests
+    .is_pct <- function(x) { tmp <- asNumericVector(x); return(!is.null(attr(tmp, "statistic"))) }
```
Comment (JustinCCYap) [inline R/ParseUserEnteredTable.R]: Should the function check that the statistic is/contains "%"?

File: R/ParseUserEnteredTable.R (hunk 1 above)
Comment (JustinCCYap) [reply R/ParseUserEnteredTable.R]: Perhaps the first `.is_pct` definition can be removed

File: R/asnumeric.R [hunk 3]
```diff
@@ -172,30 +172,23 @@ asNumericVector <- function(t)
 {
     v <- gsub(",", "", TrimWhitespace(as.vector(t)))
     v <- gsub("^\\$", "", v)
-    result <- suppressWarnings(as.numeric(v))
+
+    # Convert parentheses to negative numbers
+    patt <- "^\\(\\$?[0-9.]+%?)$"
+    is.neg <- regexpr(patt, v) > 0
```
Comment (JustinCCYap) [inline R/asnumeric.R]: Could use `grepl`?

Comment (chschan) [reply R/asnumeric.R]: Yep, I've replaced it with `grepl` in eb80adb

File: R/ParseUserEnteredTable.R (hunk 2 above)
Comment (chschan) [reply R/ParseUserEnteredTable.R]: Currently, asNumericVector only attaches an attribute if its "%", but it is safer to check for "%". Updated in eb80adb

File: R/ParseUserEnteredTable.R (hunk 1 above)
Comment (chschan) [reply R/ParseUserEnteredTable.R]: Thanks for picking that up. Removed the first one in eb80adb

**Pull request merged**